users/haervig/haervig.h

```diff
@@ -0,0 +1,305 @@
+#include "quantum.h"
+#include "keymap_danish.h"
+
+enum userspace_layers {
+  _DEADKEY = 14,            // Change if more than 16 layers are required
+  _NAV
+};
+
+enum userspace_custom_keycodes {
+  CU_GAME = SAFE_RANGE,     // Toggle game mode on/off
+  CU_NAV,                   // NAV | ESC
+
+  #ifdef DANISH_ENABLE
+  CU_LSFT,                  // LSFT | (
+  CU_RSFT,                  // LSFT | )
+  CU_COMM,                  // , | <
+  CU_DOT,                   // . | >
+  CU_SLSH,                  // / | ?
+  CU_SCLN,                  // ; | :
+  CU_QUOT,                  // ' | "
+  CU_2,                     // 2 | @
+  CU_4,                     // 4 | $
+  CU_6,                     // 6 | ^
+  CU_7,                     // 7 | &
+  CU_8,                     // 8 | *
+  CU_9,                     // 9 | (
+  CU_0,                     // 0 | )
+  CU_MINS,                  // - | _
+  CU_EQL,                   // = | +
+  CU_BSPC,                  // backspace | delete
+  CU_LBRC,                  // [ | {
+  CU_RBRC,                  // ] | }
+  CU_BSLS,                  // \ | |
+  #endif
+
+  NEW_SAFE_RANGE            // Use for keymap specific keycodes
+};
+
+#ifdef DANISH_ENABLE
+extern bool lshift;
+extern bool rshift;
+extern bool lshiftp;
+extern bool rshiftp;
+extern uint16_t lshift_timer;
+extern uint16_t rshift_timer;
+
+extern uint8_t prev_indx;
+extern uint16_t prev_kcs[6];
+void add_to_prev(uint16_t kc);
+void unreg_prev(void);
+
+extern bool esct;
+#endif
+
+extern bool navesc;
+extern uint16_t navesc_timer;
+
+extern bool game;
+
+void timer_timeout(void);
+
+void timer_timeout_keymap(void);
+
+bool process_record_keymap(uint16_t keycode, keyrecord_t *record);
+
+#define CTRLX LCTL(KC_X)
+#define CTRLC LCTL(KC_C)
+#define CTRLV LCTL(KC_V)
+
+#define ALTF4 LALT(KC_F4)
+
+#define GUIU LGUI(KC_UP)
+#define GUID LGUI(KC_DOWN)
+#define GUIL LGUI(KC_LEFT)
+#define GUIR RGUI(KC_RIGHT)
+
+#define CTLENT CTL_T(KC_ENT)
+
+#define EMOJI LWIN(KC_DOT)
+
+/*
+Templates for Keys, with custom shifted and non shifted Characters
+*/
+
+// Normal shift status
+#define SHIFT_NORM(kc1, kc2) \
+if (record->event.pressed) { \
+  timer_timeout(); \
+  if (lshift || rshift) { \
+    register_code(KC_LSFT); \
+    unregister_code(kc2); \
+    register_code(kc2); \
```

Comment:
Pairs like this can be written as:
```suggestion
tap_code(kc2);
```
If you're expecting modified keycodes here as well, use `tap_code16()`.

Reply:
Looks like I misread, actually, this won't work since the unregister is coming before the register.